self-api/src/main/java/com/selfxdsd/api/Invoices.java

```diff
@@ -0,0 +1,64 @@
+package com.selfxdsd.api;
+
+import java.time.Duration;
+import java.util.List;
+
+/**
+ * Invoices of a contract.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.3
+ * @todo #163:30min Implement and test InMemoryInvoices.
+ *  Also test StoredInvoice and StoredContract#invoices().
+ */
+public interface Invoices extends Iterable<Invoice> {
+
+    /**
+     * Find all invoices of a contract.
+     * @param id Contract's id
+     * @return Iterable of invoices.
+     */
+    Invoices ofContract(Contract.Id id);
+
+    /**
+     * Adds a completed task to the available active contract invoice.
+     * If there is no active invoice, a new invoice will be created.
+     * <br>
+     * Task criteria:
+     * <ul>
+     *     <li>Must be assigned to contract's contributor.</li>
+     *     <li>Must be part of the contract's project tasks.</li>
+     *     <li>Must not be previously added to other invoices
+     *     in the same contract, or to the active invoice.</li>
+     * </ul>
+     *
+     * @param task Completed task.
+     * @param timeSpent Time spent to finish the task.
+     * @return Created InvoiceTask
+     */
+    InvoiceTask add(Task task, Duration timeSpent);
+
+    /**
+     * Checks if invoice is active.
+     * @param id Invoice id.
+     * @return Boolean.
+     */
+    boolean isActive(int id);
+
+    /**
+     * Proceed to pay the contract's contributor.
+     * After payment confirmed, invoice will became inactive.
+     * @param id Invoice id.
+     * @return Inactive Invoice.
+     */
+    Invoice deactivate(int id);
```

Comment:
@criske I would remove this method for now. In principle it is ok, but it's too early for it, I have no idea how the payment mechanism will work yet :D